applications/Unity.GrantManager/src/Unity.GrantManager.Application/Locality/SectorAppService.cs

```diff
@@ -15,16 +20,39 @@ public class SectorAppService : ApplicationService, ISectorService
         private readonly ISectorRepository _sectorRepository;
 
         private readonly ISubSectorRepository _subSectorRepository;
+        private readonly ISettingManager _settingManager;
+        private readonly IDistributedCache<IList<SectorDto>, CacheKey> _cache;
+        private readonly ICurrentTenant _currentTenant;
+
+        private readonly string _filterType = "SectorFilter";
 
         public SectorAppService(ISectorRepository sectorRepository,
-            ISubSectorRepository subSectorRepository)
+            ISubSectorRepository subSectorRepository,
+            ISettingManager settingManager,
+            ICurrentTenant currentTenant,
+            IDistributedCache<IList<SectorDto>, CacheKey> cache)
         {
             _sectorRepository = sectorRepository;
             _subSectorRepository = subSectorRepository;
+            _settingManager = settingManager;
+            _currentTenant = currentTenant;
+            _cache = cache;
         }
 
         public async Task<IList<SectorDto>> GetListAsync()
         {
+            var cacheKey = new CacheKey { CacheType = _filterType, TenantGuid = _currentTenant.GetId() };
+#pragma warning disable CS8603 // Possible null reference return.
```

Comment:
think this can be replaced with : 
 return await _cache.GetOrAddAsync(
                cacheKey,
                GetTenantSectors()
            ) ?? new List<SectorDto>();
and no need for the pragma

Reply:
pushed now a new code